flow/beta/cli/config/json/account.go

```diff
@@ -0,0 +1,200 @@
+					Type:     key.Type,
+					Index:    key.Index,
+					SigAlgo:  crypto.StringToSignatureAlgorithm(key.SigAlgo),
+					HashAlgo: crypto.StringToHashAlgorithm(key.HashAlgo),
+					Context:  key.Context,
+				}
+				keys = append(keys, key)
+			}
+
+			account = config.Account{
+				Name:    accountName,
+				ChainID: transformChainID(a.Advanced.Chain, a.Advanced.Address),
+				Address: transformAddress(a.Advanced.Address, a.Advanced.Chain),
+				Keys:    keys,
+			}
+		}
+
+		accounts = append(accounts, account)
+	}
+
+	return accounts
+}
+
+// transformToJSON transforms config structure to json structures for saving
+func (j jsonAccounts) transformToJSON(accounts config.Accounts) jsonAccounts {
+	jsonAccounts := jsonAccounts{}
+
+	for _, a := range accounts {
+		// if simple
+		//REF: horible implementation improvemenet needed
+		if len(a.Keys) == 1 && a.Keys[0].Index == 0 &&
+			a.Keys[0].Type == config.KeyTypeHex &&
+			a.Keys[0].SigAlgo == crypto.ECDSA_P256 &&
+			a.Keys[0].HashAlgo == crypto.SHA3_256 {
+			jsonAccounts[a.Name] = jsonAccount{
+				Simple: jsonAccountSimple{
+					Address: a.Address.String(),
+					Chain:   a.ChainID.String(),
+					Keys:    a.Keys[0].Context["privateKey"],
+				},
+			}
+		} else { // if advanced
+			var keys []jsonAccountKey
+			for _, k := range a.Keys {
+				keys = append(keys, jsonAccountKey{
+					Type:     k.Type,
+					Index:    k.Index,
+					SigAlgo:  k.SigAlgo.String(),
+					HashAlgo: k.HashAlgo.String(),
+					Context:  k.Context,
+				})
+			}
+
+			jsonAccounts[a.Name] = jsonAccount{
+				Advanced: jsonAccountAdvanced{
+					Address: a.Address.String(),
+					Chain:   a.ChainID.String(),
+					Keys:    keys,
+				},
+			}
+		}
+	}
+
+	return jsonAccounts
+}
+
+type jsonAccountSimple struct {
+	Address string `json:"address"`
+	Keys    string `json:"keys"`
+	Chain   string `json:"chain"`
+}
+
+type jsonAccountAdvanced struct {
+	Address string           `json:"address"`
+	Chain   string           `json:"chain"`
+	Keys    []jsonAccountKey `json:"keys"`
+	//TODO: define more properties
+}
+
+type jsonAccountKey struct {
+	Type     config.KeyType    `json:"type"`
+	Index    int               `json:"index"`
+	SigAlgo  string            `json:"signatureAlgorithm"`
+	HashAlgo string            `json:"hashAlgorithm"`
+	Context  map[string]string `json:"context"`
+}
+
+type jsonAccount struct {
+	Simple   jsonAccountSimple
+	Advanced jsonAccountAdvanced
+}
+
+func (j *jsonAccount) UnmarshalJSON(b []byte) error {
+
+	// try simple format
+	var simple jsonAccountSimple
+	err := json.Unmarshal(b, &simple)
+	if err == nil {
+		j.Simple = simple
+		return nil
+	}
+
+	// try advanced format
+	var advanced jsonAccountAdvanced
+	err = json.Unmarshal(b, &advanced)
+	if err == nil {
+		j.Advanced = advanced
+		return nil
+	}
+
+	return err
+}
+
+func (j jsonAccount) MarshalJSON() ([]byte, error) {
+	if j.Simple.Address != "" {
+		return json.Marshal(j.Simple)
+	} else {
+		return json.Marshal(j.Advanced)
+	}
```

Comment:
```suggestion
	if j.Simple.Address != "" {
		return json.Marshal(j.Simple)
	}
	
	return json.Marshal(j.Advanced)
```

It's a bit more idiomatic in Go to just collapse the `else` case in situations like this.